### PR TITLE
multi-ns kotsapp example

### PR DESCRIPTION
**Pull request by dexhorthy**

### CODE REVIEW

**Review by austinchambers — COMMENTED** [review]
I'm thinking that this might be better as an addition to the operators kots app, which already has to work across namespaces. We could just add the cross-namespace status informers from here and make it demonstrate equivalent functionality. 

If we think there's merit to having a separate multi-namespace example, then possibly it'd be good to show another situation where this is helpful (like with operators, it's required because kudo CRDs need to be in a specific namespace, maybe there's another situation where multi-namespace is specifically required with a particular product/technology). 

Also, I think that we shouldn't recommend hardcoded namespaces where it can be avoided. I'm thinking it'd be better to have that as a config. 

Possibly we can also remove the preflight/nginx stuff from this example, as it doesn't presently seem related to multi-namespaces.

**Comment by dexhorthy** [comment]
>I'm thinking that this might be better as an addition to the operators kots app, which already has to work across namespaces. We could just add the cross-namespace status informers from here and make it demonstrate equivalent functionality.

I want a simple application that demonstrates how to use features. I don't want folks to have to wade through a bunch of operator-specific YAML to grok the basics of the namespace feature. 

> If we think there's merit to having a separate multi-namespace example, 
I do
> then possibly it'd be good to show another situation where this is helpful (like with operators, it's required because kudo CRDs need to be in a specific namespace, maybe there's another situation where multi-namespace is specifically required with a particular product/technology).

Again, don't want to complicate this.

> Also, I think that we shouldn't recommend hardcoded namespaces where it can be avoided. I'm thinking it'd be better to have that as a config.

This is a good point. Per [our advice](https://kots.io/vendor/guides/production-ready/#namespaces), even having it as a config is probably bad, but I think you're right that we should add a note to this app's readme to point out its shortcomings and to recommend against this practice.

> Possibly we can also remove the preflight/nginx stuff from this example, as it doesn't presently seem related to multi-namespaces.

We need *something* to deploy to the other namespace, and nginx seemed the simplest thing. If we're keeping it as minimal as possible though, I don't see why I couldn't remove the preflight checks if you think it helps clarity.

**Comment by austinchambers** [comment]
Sometimes it's better to do some wading if it helps provide context around making a thing production ready. In any case, I just want us to consider whether it's better to merge or it's better to create a standalone. In this case, I'm inclined to agree with you. 

As for the namespace bit, our recommendation is to use a single namespace and if that's not possible, to enable the customer to configure the namespace. Thus, if we're showing a multi-namespace example, I strongly feel that we should do that and not hardcode in the namespace. 
These examples will be copied, so I feel we should always strive to have content in them that is production ready. 

As for the preflights/support bundles, I don't see an issue with them. I just wanted to mention them in case you didn't intend for them to be there. 

As for my comments about having a use-case for multi-namespaces, I was just considering whether there were any applications/charts where that is required. For example, maybe there's a persistence helm chart that requires it. In that case, it'd maybe make a better example than nginx. If not, I think nginx is fine.

File: multi-namespace/manifests/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: nginx-hardcoded-ns
+  namespace: hardcoded
```
Comment (aj-jester) [inline multi-namespace/manifests/deployment.yaml:5]: How is this `hardcoded` namespace resource creted?

**Review by aj-jester — APPROVED** [review]

**Comment by dexhorthy** [comment]
closing as needs more discussion

**Pull request closed without merging**